### PR TITLE
add support for cluster as vpc peering target

### PR DESCRIPTION
**Pull request by jfchevrette**
This updates the terraform_vpc_peerings integration to support targetting clusters as VPC peering targets

### CODE REVIEW

File: reconcile/terraform_vpc_peerings.py [hunk 1]
```diff
@@ -32,6 +53,87 @@ def fetch_desired_state(settings):
...
+                      "'network-mgmt' access level on the cluster"
+                return None, msg
+
+            # find role to use for aws access
+            awsAccount['assume_role'] = \
+                ocm.get_aws_infrastructure_access_terraform_assume_role(
+                    cluster,
+                    awsAccount['uid'],
+                    awsAccount['terraformUsername']
+                )
```
Comment (rporres) [inline reconcile/terraform_vpc_peerings.py:102]: Is there a possibility that this returns empty? It shouldn't be the case as we should always have a `terraform` user in the cluster's access list... Maybe too defensive to check for it, just worth mentioning :)

Comment (jfchevrette) [reply reconcile/terraform_vpc_peerings.py:102]: The function should always return a string according to https://github.com/app-sre/qontract-reconcile/blob/ba004224abea6d3b0834a4a353c9873118957a96/utils/ocm.py#L165-L167

Comment (rporres) [reply reconcile/terraform_vpc_peerings.py:102]: well that `return` is inside a for loop with two  conditions with `continue` statements. I guess there's a chance that it doesn't return anything...

Comment (jfchevrette) [reply reconcile/terraform_vpc_peerings.py:102]: I see what you mean! I've added a test to check if returned value is None (if it never reaches the return statement)

**Pull request merged**